packages/itwin/tree-widget/src/test/trees/models-tree/ModelsTreeFiltering.test.ts

```diff
@@ -1051,21 +1007,27 @@ describe("Models tree", () => {
             modelId: model.id,
             categoryId: category.id,
           });
-          return {
-            rootSubject,
-            model,
-            category,
-            ...createFilteredGroupingNodeTestInfo({
-              groupedElements: [testElement1, testElement2],
-              modelId: model.id,
-              categoryId: category.id,
-              pathUntilGroupingNode: [rootSubject, model, category],
-              parentType: "category",
-            }),
-          };
+          const pathUntilTargetElement = [rootSubject, model, category];
+          const groupingNode = createClassGroupingHierarchyNode({
+            className: testElement1.className,
+            modelId: model.id,
+            categoryId: category.id,
+            elements: [testElement1.id, testElement2.id],
+            parentKeys: pathUntilTargetElement.map((key) => ({ type: "instances", instanceKeys: [key] })),
+          });
+          return { rootSubject, model, category, pathUntilTargetElement, groupingNode };
         },
-        (x) => x.filterPaths,
-        (x) => [x.filterTargetItem],
+        (x) =>
+          x.groupingNode.groupedInstanceKeys.map((elementKey) => ({
+            path: [...x.pathUntilTargetElement, elementKey],
+            options: { autoExpand: { key: x.groupingNode.key, parentKeysCount: x.groupingNode.parentKeys.length } },
```

Comment:
These tests were already testing cases for focusing on a grouping node. See previous focused item definition for this test:
```(x) => [{ parent: { type: "category", ids: [x.category.id], modelIds: [x.model.id] }, classes: [x.testElement1.className] }],```

You can see that expectation for `autoExpand === true` for parent nodes was already there before my changes

Reply:
Yes, we were checking that the hierarchy is correctly auto-expanded up to the target node. What I question is whether we should change the auto-expand option from `true` to the grouping node. That essentially changes the test, so if you think there's value in testing that, then I believe that should be a separate test.

Reply:
I don't see, how this changes the test. The test was already targeting a grouping node. See focused item before the changes:
```ts
 (x) => [{ parent: { type: "category", ids: [x.category.id], modelIds: [x.model.id] }, classes: [x.testElement1.className] }],
 ```

Reply:
OK, sorry about the confusion. I now see these tests were specifically added for testing grouping nodes' focusing. In that case, it would be great to rename the tests similarly to how I described [here](https://github.com/iTwin/viewer-components-react/pull/1010#discussion_r1714711254):
- "grouping element node under category" I think can be removed as it's covered with the test below
- "grouping element node under category and correct model" -> "grouped root element, when expanding up to its grouping node"
- "grouping element node under parent element" -> "grouped child element, when expanding up to its grouping node"
- "multiple grouping element nodes under parent element" -> "grouped child elements of different classes, when expanding up to their grouping nodes"

I guess, if we move all these tests under a `describe("when expanding up to element class grouping nodes", ...)`, then the suffix can be removed from the test names.

Reply:
Gotcha

Reply:
Moved these test cases under a separate `describe` and renamed accordingly